frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -47,20 +47,64 @@ interface PublicFormProviderProps {
   children: React.ReactNode
 }
 
+export function useCommonFormProvider(formId: string) {
+  const [vfnTransaction, setVfnTransaction] =
+    useState<FetchNewTransactionResponse>()
+  const miniHeaderRef = useRef<HTMLDivElement>(null)
+  const { createTransactionMutation } = useTransactionMutations(formId)
+  const toast = useToast({ isClosable: true })
+  const vfnToastIdRef = useRef<string | number>()
+  const desyncToastIdRef = useRef<string | number>()
+
+  const getTransactionId = useCallback(async () => {
+    if (!vfnTransaction || isPast(vfnTransaction.expireAt)) {
+      const result = await createTransactionMutation.mutateAsync()
+      setVfnTransaction(result)
+      return result.transactionId
+    }
+    return vfnTransaction.transactionId
+  }, [createTransactionMutation, vfnTransaction])
```

Comment:
this might fail in preview mode, since the form could be private. we should have distinct functions between preview and public mode for verification

Reply:
Is the reason for distinct functions because this might fail at `createTransactionMutation`? Traced it to [this](https://github.com/opengovsg/FormSG/blob/form-v2/develop/src/app/modules/verification/verification.controller.ts#L32), from the APIs called it looks like there's no check on whether the form is public or private, just whether the formId exists in the DB.

Relatedly, how do we want the behaviour of the functions to be different for preview vs public verification?

Reply:
As discussed - to maintain feature parity with the current Angular version, will leave the current PR as it is (same function used for both public and private forms)

Will open a new issue for mocking transactions in preview mode